### DISCUSSION TITLE
Clarity on what Upscayl Cloud actually is

### DISCUSSION THREAD

**Discussion by TheOverpassArsonist** [upscayl/upscayl#500] [Q&A]
### Checklist

- [X] I have checked that this issue isn't addressed in [the FAQ](https://github.com/upscayl/upscayl#faq-).
- [X] I have checked that this issue isn't addressed in any other issue or pull request.

### Describe the Issue

Upscayl Cloud is advertised a bit, but the specific features it's advertised with raise some questions that I haven't been able to find any answers on. Specifically a few points are listed frequently,

- Upscayl anywhere, anytime, any device
Reasonable, it's a cloud service it would work anywhere.

- No GPU or hardware required
Reasonable, it's a cloud service so you don't need good hardware

- Face Enhancement
Is this something unique to the cloud service? If so, what tool is being used? If it's in-house but not in the desktop app is that because it wouldn't run well, or is the cloud service closed source and has functionality intentionally unique from the desktop app?

- 10+ models to choose from
Reasonable, it's a cloud service so you can include several models that couldn't be reasonably shipped included with every user-installation. (I'm assuming these models aren't unique to the cloud service and can be added manually if a user wants to use them locally)

- 5x faster than Upscayl Desktop
Reasonable, it's a cloud service so it can run on way stronger hardware to go faster

- Video Upscaling
Similar to face enhancement; this feature isn't in the desktop version (at least not the flatpak) at all as far as I can tell, so is it going to be unique to the cloud service? If so then that raises the same licensing questions as before. Video upscaling is something that can be done locally, so other than having to redesign the UI to accomodate it I can't think of any reason that it wouldn't be implemented in the desktop version unless it was under a different license. 

- Commercial Usage
I'm admittedly not a lawyer but I've looked over the AGPL-3.0 and it seems to allow commercial usage, even the github summary at the top marks it as allowed. (image below) I suppose I can rationalize this as being listed in reference to **_other_** competing cloud upscaling services which may **_not_** allow for commercial usage, but it's still odd given this same item also appears in the pop-up within the Upscayl desktop application itself. Presumably people using the desktop Upscayl application wouldn't be comparing Upscayl Cloud with other cloud upscaling services, they'd be comparing it with the Upscayl desktop application, so adding this point within the desktop prompt implies you can't use the desktop application comercially despite the license seeming to suggest you can. I could maybe see this meaning that "it's designed so that you can integrate it into an automated service", so a bit like how you can buy AWS servers to host your content you could use Upscayl Cloud to upscale content automatically, but again that would just make me ask why it's being listed as a feature when a user opens the Upscayl desktop application. The users who would care about being able to easily pipeline dozens or hundreds or thousands of automated upscale requests to cloud servers don't really seem like they'd share a lot of overlap with the average user just wanting to upscale an old family photo.

Overall there is just a lot of strange phrasing and absent features surrounding Upscayl Cloud that I can't find any information on. Is Upscayl Cloud also open source, or is it using Upscayl's open source components with closed source extensions? Is the point about commercial usage just strange phrasing or are you not allowed to use the desktop Upscayl desktop application commercially? If it Upscayl Cloud isn't closed source and will be relyong the same underlying technology as the desktop application then why would features like video upscaling be exclusive to the cloud service? Maybe I've missed something but I've checked everywhere I can think to and couldn't find anything regarding differences in licensing, functionality, etc. between Upscayl and Upscayl Cloud so, unless I've just been a massive idiot and it's super obvious, I think it's at least worth making this information more clearly communicated in the case that it does exist and I've just missed it.

### Screenshots

![image](https://github.com/upscayl/upscayl/assets/142986657/b285bc50-3db2-4ad8-95ab-376edeb19ca7)
![image](https://github.com/upscayl/upscayl/assets/142986657/f3a99f92-cbbe-43b2-9b68-eac6f41a3307)

edit : if it needs to be said, I'm not against offering a cloud service here, and I'm not necessarily even against how it's being advertised in general, but there are aspects of it which are unclear and have some rather bad implications from how they're phrased. I see absolutely nothing wrong with a cloud service offering similar features with better performance, more platform agnosticism, etc. because that's just the nature of paying to use a cloud service; those are genuine value-adds intrinsically linked to that concept. However, the phrasing of these bullet points imply (to varying extents) something entirely different, that being that the cloud service is closed source, there could be potential license changes to restrict the use of the desktop version, there may be intentional feature exclusivity between the local and cloud versions, etc. I'm not saying any of those things are true and I can't find any hard evidence they are, but I also can't find any evidence that they're not, so it _**is**_ definitely concerning.

**Comment by NayamAmarshe** [comment]
Upscayl Cloud will not be using the same tech as Upscayl Desktop, that is the biggest difference.

To be very honest, we're still in the early stage. We have to evaluate several options before we can confidently have a concrete offering. What the website mentions is just a rough overview of what you can expect, of course things may change in due course of time.

Desktop version won't change in any way of course. The cloud version is just for people who either want faster upscaling or don't have the necessary hardware to use the desktop version.

The licensing and everything else will be more clear (for us and you all) later when we rollout the beta :)

Let me know if you have any other questions. Thank you! :D

**Comment by shenef** [comment]
Make sure to add a privacy policy to the site.
Per the GDPR, you are supposed to disclose how collected data is being processed if you collect the data of EU citizens and currently that is not known.
(unless you want to exclude EU users, but that defeats the "Upscayl anywhere, anytime, any device")

**Comment by NayamAmarshe** [comment]
> Make sure to add a privacy policy to the site. Per the GDPR, you are supposed to disclose how collected data is being processed if you collect the data of EU citizens and currently that is not known. (unless you want to exclude EU users, but that defeats the "Upscayl anywhere, anytime, any device")

Yeah, those things are for later, for the most part.
Currently, it's about getting a solid service up and running :)

**Reply by Patrerick** [reply]
Has a Privacy Policy been published yet? If so, where can we find it?
